### ISSUE TITLE
[Feature] exclude / filter domains from results (client side)

### ISSUE THREAD

**Issue by fuck-shithub** [searxng/searxng#2351]
Update from the maintainers: there is a POC based on [Window: localStorage property](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage)

-  https://github.com/searxng/searxng/pull/2946

-----

Feature request for excluding/filtering domains from the results on the client side.

Searx related issue: https://github.com/searx/searx/issues/2001

-----

**Is your feature request related to a problem? Please describe.**
Often when you search, engines like Google and DDG return completely unhelpful results that just shill their own product or don't answer the question or query you were searching for.

**Describe the solution you'd like**
The possibility for users to blacklist a website from ever showing up in results again and this gets saved in cookies. Would be great if you can also change the list of blocked websites in preferences.

**Describe alternatives you've considered**
Maybe instance-wide blocking of websites decided by the admins of the instance. Or a suggested blacklist the user can opt-in, or opt-out of, or modify at will.
Another expansion upon this idea would be a community-maintained blacklist of all SEO/AI-generated/unhelpful websites so we can all put an end to this once and for all.

**Comment by unixfox** [comment]
@return42 I think there is already an existing issue for the hostname_replace plugin but on the user side? I can't find it.

**Comment by return42** [comment]
Yes I remember we had a discussion but in context of "client side" I can find only this comment:

- https://github.com/searxng/searxng/issues/1086#issuecomment-1096983229
- https://github.com/searxng/searxng/issues/1575

> The possibility for users to blacklist a website from ever showing up in results again and this gets saved in cookies. Would be great if you can also change the list of blocked websites in preferences.

The cookie size is limited, see https://github.com/searxng/searxng/issues/1575#issuecomment-1198978876

> Maybe instance-wide blocking of websites decided by the admins of the instance. 
> ... Another expansion upon this idea would be a community-maintained blacklist of all SEO/AI-generated/unhelpful websites so we can all put an end to this once and for all.

We don't have a solution yet where you can enable or disable a block list.  In the long term, I could imagine something like that. Maybe something can arise from this discussion

- https://github.com/searxng/searxng/discussions/970

For those are coming here I would like to propose to participate in the discussion linked above.

**Comment by unixfox** [comment]
Searx related issue: https://github.com/searx/searx/issues/2001

**Comment by tokenwizard** [comment]
I, too would like to block/filter certain domains from showing in the SearxNG results. I have seen lots of threads that mention the hostname_replace plugin and have mine setup like this:

```
enabled_plugins:
  - 'Hostname replace'

hostname_replace:
# block bload domains
  '(.*\.)?medium.com': false
  '(*\.)?uxdesign\.cc': false
  '(.*\.)?medium\.com$': false
  '*\.medium\.com': false
  'com\.de': false
  '(.*\.)?codegrepper\.com': false
  '(.*\.)?w3schools\.com': false
  '(.*\.)?geeksforgeeks\.org': false
  '(.*\.)?stackshare\.io': false
  '(.*\.)?tutorialspoint\.com': false
  '(.*\.)?answeright\.com': false
  '(.*\.)?askdev\.info': false
  '(.*\.)?askdev\.io': false
  '(.*\.)?blogmepost\.com': false
  '(.*\.)?c-sharpcorner\.com': false
  '(.*\.)?code-examples\.net': false
  '(.*\.)?codeflow\.site': false
  '(.*\.)?gitmemory\.cn': false
  '(.*\.)?gitmemory\.com': false
  '(.*\.)?intellipaat\.com': false
  '(.*\.)?javaer101\.com': false
  '(.*\.)?programmerstart\.com': false
  '(.*\.)?programmersought\.com': false
  '(.*\.)?qastack\.com': false
  '(.*\.)?roboflow\.ai': false
  '(.*\.)?stackanswers\.net': false
  '(.*\.)?stackoom\.com': false
  '(.*\.)?stackovernet\.com': false
  '(.*\.)?stackovernet\.xyz': false
  '(.*\.)?stackoverrun\.com': false
  '(.*\.)?thetopsites\.net': false
  '(.*\.)?ubuntugeeks\.com': false
  '(.*\.)?cyberciti\.biz': false
  '(.*\.)?ispycode\.com': false
  '(.*\.)?reposhub\.com': false
  '(.*\.)?githubmemory\.com': false
  '(.*\.)?issueexplorer\.com': false
  '(.*\.)?tabnine\.com': false
  '(.*\.)?gitcode\.net': false
  '(.*\.)?command-not-found\.com': false
  '(.*\.)?im-coder\.com': false
  '(.*\.)?i-harness\.com': false
  '*\.medium\.com': false

  # ublock-origin-shitty-copies-filter/stackoverflow-copycats.txt

  '(.*\.)?code-examples\.net': false
  '(.*\.)?kotaeta\.com': false
  '(.*\.)?answer-id\.com': false

```

The ublock list goes on quite a bit further. Unfortunately, I still get sh*^$% medium.com and other bulk$&^%$ results in my searches:

![image](https://github.com/searxng/searxng/assets/24832561/a9363454-4482-4a07-89da-2af635e859f3)

I'm sure my regex is off, or there's something I'm missing.

**Comment by return42** [comment]
> The possibility for users to blacklist a website from ever showing up in results again and this gets saved in cookies. 

I close this feature request in favorite of https://github.com/searxng/searxng/issues/2351#issuecomment-1510826350

**Comment by ghost** [comment]
Can it be done indirectly by adding parameters to certain search engines? E.g. DuckDuckGo allows filtering out sites with `Wikipedia -site:wikipedia.org`. It does affect the pagination and not return the same amount of results. It filters it out.

**Comment by return42** [comment]
> Can it be done indirectly by adding parameters to certain search engines? 

No, that's not possible, simply because some of the pages queried by the engines either don't have pagination or their API dictates a fixed pagination. Please remember that most sources (Google, DDG, etc.) don't support automated queries, and the engines are always implemented to extract the maximum amount of data.
